### PR TITLE
Feature obsreq 104 datalist component

### PR DESCRIPTION
**Pull request by akate1**
**Description:**
Added new datalist component
https://clarity.design/documentation/datalist#examples

**Screen-shots:**
image

image

image

image

### CODE REVIEW

File: src/forms/datalist/DataList.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+// Datalist option component
+export const DataListOption: React.FunctionComponent<DataListOptionProps> = ({
+    value,
+    selected,
+    children,
+    clasName,
+    style,
+}) => {
+    return (
+        <option value={value} selected={selected} className={ClassNames.NG_STAR_INSERTED} style={style}>
```
Comment (maheshkulkarni4) [inline src/forms/datalist/DataList.tsx]: Whats the use of NG_STAR_INSERTED here ?

File: src/forms/datalist/DataList.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+// DataList component
+export class DataList extends React.PureComponent<DataListProps, DataListState> {
+    constructor(props: DataListProps) {
+        super(props);
+        this.state = {
+            hasFocus: false,
+        };
+    }
+
+    private handleFocus = (evt: React.FocusEvent<HTMLInputElement>) => {
```
Comment (maheshkulkarni4) [inline src/forms/datalist/DataList.tsx]: evt can be empty, we are not using it anyway.  Should we add evt.preventDefault for avoding bubbling of this event to parent ?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@dell/clarity-react",
-    "version": "0.21.1",
+    "version": "0.21.2",
```
Comment (psivakumaran) [inline package.json]: Should this be 0.22.0?

File: src/forms/datalist/index.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (psivakumaran) [inline src/forms/datalist/index.ts]: 2018 -> 2019

File: src/forms/datalist/ClassNames.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (psivakumaran) [inline src/forms/datalist/ClassNames.tsx]: 2018 -> 2019
Same for other files

File: src/forms/datalist/DataList.tsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+type DataListProps = {
+    placeHolder?: string;
+    label?: string;
+    name?: string;
+    required?: boolean;
+    isError?: boolean;
+    errorText?: string;
+    helperText?: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+    onBlur?: (evt: React.FocusEvent<HTMLInputElement>) => void;
```
Comment (psivakumaran) [inline src/forms/datalist/DataList.tsx]: Should this be onFocus?

File: src/forms/datalist/DataList.tsx (hunk 1 above)
Comment (akate1) [reply src/forms/datalist/DataList.tsx]: its a CSS class name :)

File: src/forms/datalist/DataList.tsx (hunk 6 above)
Comment (akate1) [reply src/forms/datalist/DataList.tsx]: we need to perform some actions after onBlur() so keeping it as it is.

File: src/forms/datalist/DataList.tsx (hunk 2 above)
Comment (akate1) [reply src/forms/datalist/DataList.tsx]: @maheshkulkarni4  I don't think evt will be empty here. Can you please let me know any such scenario that may be I am missing ?

File: package.json (hunk 3 above)
Comment (akate1) [reply package.json]: @maheshkulkarni4 and @ben should we update minor version here or major ? Please provide your thoughts.

Comment (ben-schumacher) [reply package.json]: I guess we are adding functionality -I'll bump to 0.22.0.

**Pull request merged**